File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.cpp
class Solution 
{
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) 
    {
        //sort using lamda function
        sort(points.begin(), points.end(), [](const vector<int> a, const vector<int> b)
                                            {
                                               // int x = (a[0] * a[0] + a[1] * a[1]);
                                               // int y = (b[0] * b[0] + b[1] * b[1]);
                                               // cout << x <<"  "<< y<<endl;
                                               // return x < y;

                                                return (a[0] * a[0] + a[1] * a[1]) < (b[0] * b[0] + b[1] * b[1]);
                                            }
        );
        return vector<vector<int>>(points.begin(), points.begin() + k);
    }
};

File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.py
class Solution(object):
    def kClosest(self, points, k):
        """
        :type points: List[List[int]]
        :type k: int
        :rtype: List[List[int]]
        """

        points.sort(key=lambda p: hypot(p[0], p[1]))
        return points[:k]